src/redfish_collector/core/rawCollector.py

```
@@ -55,40 +57,43 @@ def dataJSONWriter(dataRaw,fileDir,fileName, serverAddress):
-            raise
+    async with semaphore:
+        for attempt in range(1, retries + 1):
+            try:
+                async with session.get(url, headers=headers, ssl=False) as response:
+                    response.raise_for_status() 
+                    return await response.json()
+            except (ClientConnectorError, ClientResponseError, TimeoutError, OSError) as e:
+                logging.debug("[%s] Attempt %s with url: %s failed: %s", serverAddress, attempt, url, e)
+                
+                if attempt == retries:
+                    logging.error("[%s] Max retries reached. Giving up.", serverAddress)
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)